ui/src/warp_runner/mod.rs

```diff
@@ -109,110 +122,162 @@ async fn handle_login(notify: Arc<Notify>) {
                 }
 
                 match opt {
-                Some(WarpCmd::MultiPass(MultiPassCmd::CreateIdentity {
-                    username,
-                    passphrase,
-                    rsp,
-                })) => {
-                    tesseract.clear();
-                    tokio::time::sleep(std::time::Duration::from_millis(100)).await;
-                    let new_account = true;
-                    let mut warp = match login(&passphrase, tesseract.clone(), new_account).await {
-                        Ok(w) => w,
-                        Err(e) => {
-                            let _ = rsp.send(Err(e));
-                            continue;
-                        }
-                    };
-                    match warp.multipass.create_identity(Some(&username), None).await {
-                        Ok(_id) => {
-                            // don't want a panic during the first run of the app to mess up tesseract.
-                            // calling save() here is intended to ensure that the username and password will
-                            // work the next time uplink is run.
-                            let _ = warp.tesseract.save();
-
-                            let _ = rsp.send(Ok(()));
-                            break Some(warp);
+                    Some(WarpCmd::MultiPass(MultiPassCmd::CreateIdentity {
+                        username,
+                        passphrase,
+                        rsp,
+                    })) => {
+                        if account_exists {
+                            let tesseract = init_tesseract(true)
+                                .await
+                                .expect("failed to initialize tesseract");
+                            warp = match warp_initialization(tesseract, false).await {
+                                Ok(w) => w,
+                                Err(e) => {
+                                    log::error!("warp init failed: {}", e);
+                                    return;
+                                }
+                            };
                         }
-                        Err(e) => {
-                            log::error!("create_identity failed. should never happen: {}", e);
+
+                        if let Err(e) = warp.tesseract.unlock(passphrase.as_bytes()) {
+                            log::info!("unlock failed: {:?}", e);
                             let _ = rsp.send(Err(e));
+                            continue;
+                        };
+                        match warp.multipass.create_identity(Some(&username), None).await {
+                            Ok(_id) =>  match wait_for_multipass(&mut warp, notify.clone()).await {
+                                Ok(_) => match save_tesseract(&warp.tesseract) {
```

Comment:
Any reason to save when this is done internally once the autosave flag is set?

Reply:
i was being paranoid